### PR TITLE
A Cypress-based stress tests for polls

### PR DESCRIPTION
**Pull request by andybalaam**
A test suite that doesn't run as part of our end-to-end tests, intended for larger tests that stress the system.

Includes one test that creates a poll and has 200 people vote in it.
 

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: cypress-stress.config.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    experimentalInteractiveRunEvents: true,
+    defaultCommandTimeout: 10000,
+    chromeWebSecurity: false,
+    e2e: {
+        setupNodeEvents(on, config) {
+            return require('./cypress/plugins/index.ts').default(on, config);
+        },
+        baseUrl: 'http://localhost:8080',
+        experimentalSessionAndOrigin: true,
+        specPattern: 'cypress/stress/**/*.{js,jsx,ts,tsx}',
```
Comment (weeman1337) [inline cypress-stress.config.ts:31]: To prevent JavaScript based tests from popping up, shouldn't we better just allow TypeScript here?

**Review by t3chguy — CHANGES_REQUESTED** [review]
As it stands, this is effectively dead code and could get broken at any time. We should get this running in CI at least periodically.

File: cypress/stress/polls/polls-in-large-rooms.spec.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        });
+    });
+});
+
+type CreatePollOptions = {
+    title: string;
+    options: string[];
+};
+
+const createPoll = ({ title, options }: CreatePollOptions) => {
```
Comment (weeman1337) [inline cypress/stress/polls/polls-in-large-rooms.spec.ts:139]: These could go to poll helpers / commands, so that we can reuse them in other tests.

**Review by weeman1337 — COMMENTED** [review]
Thanks for the test :+1:  Like Michael T says it would be nice to run these tests from time to time. What about scheduling them on the day before releases?

File: cypress-stress.config.ts (hunk 1 above)
Comment (andybalaam) [reply cypress-stress.config.ts]: Fixed in 095bfc616777ef05cb8fda735f1f2e68e06f002b

**Comment by t3chguy** [comment]
>  I don't know how to test this workflow without merging...

The syntax looks invalid to me, can you put this on a fork and test it there?

File: .github/workflows/cypress-stress.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    name: "Run Stress Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run Cypress tests
+        uses: cypress-io/github-action@v4.1.1
+        with:
+          browser: chrome
+          start: npx serve -p 8080 webapp
```
Comment (t3chguy) [inline .github/workflows/cypress-stress.yaml:19]: Where is `webapp` coming from?

**Review by t3chguy — COMMENTED** [review]
Can you test

**Comment by andybalaam** [comment]
> Can you test

Yeah, GitHub seems to be running this whenever I push a change, so I can at least see the stupid mistakes I make.

File: .github/workflows/cypress-stress.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          wait-on: 'http://localhost:8080'
+          record: true
+        env:
+          # pass the Dashboard record key as an environment variable
+          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
```
[the remark this answers is not in the document]
Comment (andybalaam) [reply .github/workflows/cypress-stress.yaml:24]: OK, I'll remove it for now and handle recordings later.

File: .github/workflows/cypress-stress.yaml (hunk 3 above)
Comment (andybalaam) [reply .github/workflows/cypress-stress.yaml:19]: Ha, I forgot I need to get element-web from somewhere.

**Comment by andybalaam** [comment]
Given:

- the pain I am experiencing trying to get the workflow to run,
- the fact that our github runners have highly variable performance, so it's quite likely this test will be flakey even if we get it right, and
- I really need to test this across multiple clients, not just Element Web

I am going to abandon this PR and look for a way to do something similar with https://github.com/matrix-org/trafficlight

**Pull request closed without merging**